cw/capture.py

```diff
@@ -991,6 +1008,235 @@ def kmac_fvsr_key_batch(ctx: typer.Context,
+            actual_key = (d0 + d1) % curve_order_n
+            expected_key = expected_fixed_key
+            seedout = seed_fixed
+        else:
+            # Set the seed.
+            seed = ktp.next_key()
+            ot.target.simpleserial_write('x', seed)
+            tqdm.write(f'seed   = {seed.hex()}')
+
+            # Check for errors.
+            err = ot.target.read()
+            if err:
+                raise RuntimeError(f'Error writing seed: {err}')
+
+            # Arm the scope
+            ot.scope.arm()
+
+            # Send the mask and start the keygen operation.
+            ot.target.simpleserial_write('k', mask)
+
+            # Wait until operation is done.
+            ret = ot.scope.capture(poll_done=True)
+            if ret:
+                raise RuntimeError('Timeout during capture')
+
+            # Check the number of cycles where the trigger signal was high.
+            cycles = ot.scope.adc.trig_count
+            tqdm.write("Observed number of cycles: %d" % cycles)
+
+            waves = ot.scope.get_last_trace(as_int=True)
+            # Read the output, unmask the key, and check if it matches
+            # expectations.
+            share0 = ot.target.simpleserial_read("r", seed_bytes, ack=False)
+            share1 = ot.target.simpleserial_read("r", seed_bytes, ack=False)
+            if share0 is None or share1 is None:
+                raise RuntimeError('Did not receive expected output.')
+
+            d0 = int.from_bytes(share0, byteorder='little')
+            d1 = int.from_bytes(share1, byteorder='little')
+            actual_key = (d0 + d1) % curve_order_n
+            expected_key = int.from_bytes(seed, byteorder='little') % curve_order_n
+            seedout = seed
+
+        tqdm.write(f'share0 = {share0.hex()}')
+        tqdm.write(f'share1 = {share1.hex()}')
+
+        sample_fixed = random.randint(0, 1)
+        if actual_key != expected_key:
+            raise RuntimeError('Bad generated key:\n'
+                               f'Expected: {hex(expected_key)}\n'
+                               f'Actual:   {hex(actual_key)}')
+
+        # Create a chipwhisperer trace object and save it to the project
+        # Args/fields of Trace object: waves, textin, textout, key
+        textout = share0 + share1  # concardinate bytearrays
+        trace = Trace(waves, mask, textout, seedout)
+        check_range(waves, ot.scope.adc.bits_per_sample)
+        project.traces.append(trace, dtype=np.uint16)
+
+    project.save()
+
+
+@app_capture.command()
+def otbn_vertical(ctx: typer.Context,
+                  num_traces: int = opt_num_traces,
+                  plot_traces: int = opt_plot_traces):
+    """Capture ECDSA secret key generation traces."""
+    capture_init(ctx, num_traces, plot_traces)
+
+    # OTBN's public-key operations might not fit into the sample buffer of the scope
+    # These two parameters allows users to conrol the sampling frequency
+    #
```

Comment:
Do we need this empty comment line?

Reply:
Not necessarily, but it's the same structure as in `ecdsa_simple` and `ecdsa_stream`.
Should I remove it there as well? Or better do that in a different PR?

Reply:
I was keeping them for the readability. If it violates the styling guidelines, we can remove them with the `reset_firmware` PR proposed in [this comment](https://github.com/lowRISC/ot-sca/pull/115#discussion_r1118735037) above.